packages/api-client/src/client/ClientAPI.ts

```diff
@@ -218,4 +243,40 @@ export class ClientAPI {
     const response = await this.client.sendJSON<PublicKeys>(config, true);
     return response.data;
   }
+
+  public async getNonce(clientId: string): Promise<string> {
+    const NonceKey = 'replay-nonce';
+    const NonceSchema = z.object({
+      [NonceKey]: z.string().min(1),
+    });
+
+    const config: AxiosRequestConfig = {
+      method: 'head',
+      url: `${ClientAPI.URL.CLIENTS}/${clientId}/${ClientAPI.URL.NONCE}`,
+    };
+
+    const {headers} = await this.client.sendRequest(config);
+
+    return NonceSchema.parse(headers)[NonceKey];
+  }
+
+  public async getAccessToken(clientId: string, dpopToken: Uint8Array) {
+    const ResponseSchema = z.object({
+      expires_in: z.number().min(1),
+      token: z.string().min(1),
+      type: z.literal('DPoP'),
+    });
+
+    const config: AxiosRequestConfig = {
+      method: 'post',
+      url: `${ClientAPI.URL.CLIENTS}/${clientId}/${ClientAPI.URL.ACCESS_TOKEN}`,
+      headers: {
+        DPoP: new TextDecoder().decode(dpopToken),
```

Comment:
Is this really supposed to be converted to text, not base64. 
We might lose bits if we encode it to text. Please make sure that this is the expected format and we should discuss with backend why this should be the format.

Reply:
It is supposed to be like that. I will ask backend if we want to change that

Reply:
I did ask the backend, but I would go forward and merge it as is for now. If a change is made in the future, we need to react to it anyways